2-Pricing/src/components/Cart.tsx

```diff
@@ -0,0 +1,94 @@
+import type { Cart as ICart } from "../App";
+import { Backdrop } from "./Backdrop";
+import { Button } from "./Button";
+import { Subscription, subscriptions } from "../constants";
+import { CheckoutRow } from "./CheckoutRow";
+
+export type SubscriptionKey = keyof typeof subscriptions;
+
+interface CartProps {
+  isMonthlyView: boolean;
+  inCart: ICart;
+  setIsCartOpen: (cartIsVisible: boolean) => void;
+  setInCart: (newCartContents: ICart) => void;
+}
+
+export const Cart = ({
+  isMonthlyView,
+  inCart,
+  setIsCartOpen,
+  setInCart,
+}: CartProps) => {
+  const handleCloseCart = () => setIsCartOpen(false);
+
+  const getSubscriptionByName = (
+    subscriptionName: SubscriptionKey
+  ): Subscription => subscriptions[subscriptionName];
+
+  const subscriptionsWithItemsInCart = Object.entries(inCart)
+    .filter(([subscriptionName, quantityInCart]) => quantityInCart > 0)
+    .map(([subscriptionName, quantityInCart]) => subscriptionName);
+
+  const rows = subscriptionsWithItemsInCart.map((subscriptionName) => (
+    <CheckoutRow
+      key={subscriptionName}
+      isMonthlyView={isMonthlyView}
+      inCart={inCart}
+      subscription={getSubscriptionByName(subscriptionName as SubscriptionKey)}
+      setInCart={setInCart}
+    />
+  ));
```

Comment:
I think it's better to just do the `map` inside the `return` and not declare the `rows` variable.

By doing this, you are mixing the JSX code with the logic before the return, and I think it makes sense to first have all the JS and then all the JSX inside the return.